src/expression.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any
+from .token_type import TokenType, Token
+
+
+class Expr:
+    def __call__(self) -> Any:
+        raise RuntimeError("Empty Expresson")
+
+
+class Binary(Expr):
+    left: Expr
+    operator: Token
+    right: Expr
+
+    def __init__(self, left: Expr, operator: Token, right: Expr):
+        self.left = left
+        self.operator = operator
+        self.right = right
+
+    def __str__(self) -> str:
+        return f"( {self.operator} {str(self.left)}, {str(self.right)} )"
+
+    def __call__(self) -> Any:
+        if self.operator.token_type == TokenType.STAR:
```

Comment:
We can use the `is` operator instead which check if it's the same object, might be faster and more clear.

Reply:
Idk I feel like `is` is very finicky, we can use it if u want I just don't use it cus it always fucks me over